processor/deltatocumulativeprocessor/linear.go

```diff
@@ -0,0 +1,212 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package deltatocumulativeprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor"
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/staleness"
+	exp "github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/streams"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/delta"
+	telemetry "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/lineartelemetry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/metrics"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+)
+
+var _ processor.Metrics = (*Linear)(nil)
+
+type Linear struct {
+	next processor.Metrics
+	cfg  Config
+
+	state state
+	mtx   sync.Mutex
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	stale staleness.Tracker
+	tel   telemetry.Metrics
+}
+
+func newLinear(cfg *Config, tel telemetry.Metrics, next processor.Metrics) *Linear {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	proc := Linear{
+		next: next,
+		cfg:  *cfg,
+		state: state{
+			nums: make(exp.HashMap[data.Number]),
+			hist: make(exp.HashMap[data.Histogram]),
+			expo: make(exp.HashMap[data.ExpHistogram]),
+		},
+		ctx:    ctx,
+		cancel: cancel,
+
+		stale: staleness.NewTracker(),
+		tel:   tel,
+	}
+
+	tel.WithTracked(proc.state.Len)
+	cfg.Metrics(tel)
+
+	return &proc
+}
+
+func (p *Linear) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	now := time.Now()
+
+	const (
+		keep = true
+		drop = false
```

Comment:
`drop` isn't used, right? 🤔

Reply:
not right now, but I think having it helps illustrate what those consts do. vote to keep

Reply:
I think that's a symptom of the complexity of nested functions 😬, but I agree that the constants help with the current situation! Vote to keep as well